doc/Compilation.md

```diff
@@ -129,27 +147,26 @@ in the stm32f412gdiscovery.elf kernel ELF, and creates a new .elf called
 ## Compiling a process
 
 Unlike many other embedded systems, compilation of application code is entirely
-separated from the kernel in Tock. An application is combined with at least two
-libraries: `libtock` and `newlib` and built into a free-standing binary. The
-binary can then be uploaded onto a Tock platform with an already existing
-kernel to be loaded and run.
-
-Tock can support any programming language and compiler provided they meet
-the following requirements:
-
- 1. The application must be built as position independent code (PIC).
+separated from the kernel in Tock. An application uses a `libtock` library and
+is built into a free-standing binary. The binary can then be uploaded onto a
+Tock platform with an already existing kernel to be loaded and run.
 
- 2. The application must be linked with a loader script that places Flash
-    contents above address `0x80000000` and RAM contents below it.
+Tock can support applications using any programming language and compiler
+provided the applications can run with only access to fixed regions in flash and
+RAM and without virtual memory.
 
- 3. The application binary must start with a header detailing the location of
-    sections in the binary.
+Each Tock process requires a header that informs the kernel of the size of the
+application's binary and where the location of the entry point is within the
+compiled binary.
 
-The first requirement is explained directly below while the other two are
-detailed in [Tock Binary Format](#tock-binary-format).
+### Executing without Virtual Memory
 
+Tock supports resource constrained microcontrollers which do not support virtual
+memory. This means Tock process cannot assume a known address space. Tock
+supports two methods for enabling processes despite the lack of virtual memory:
+embedded PIC (ePIC) and fixed address loading.
```

Comment:
```suggestion
Position Independent Code (PIC) and fixed address loading.
```

I don't think "ePIC" is a thing (yet?). I tried looking it up, and the only thing I could find was a [whitepaper](https://riscv-europe.org/media/proceedings/posters/2023-06-08-Hugo-MCNALLY-abstract.pdf) and [fairly inactive repo](https://github.com/luismarques/epic-spec/) talking about an ABI proposal for RISC-V. We use bog-standard PIC; all the "ePIC" changes should be reverted.

Reply:
But we don't use standard PIC because of the magic `-mno-pic-data-is-text-relative` flag.

https://github.com/riscv-collab/riscv-gcc/issues/158

Reply:
Huh, looks like "FDPIC" might be the right word based on the comment linked there, and a quick google that turns up a ton of info on FDPIC

Reply:
Ok that's right. Updated.

Reply:
I don't think we need to update the document, but this isn't quite right. FDPIC is distinct from the ePIC that lowRISC is working on, which is distinct from the type of PIC we use on ARM in `libtock-c`. I don't think there's any standard name for the type of PIC that `libtock-c` uses on ARM.

For reference, FDPIC supports shared libraries, while ePIC does not.